tests/app/views/contexts/test_section_summary_context.py

```diff
@@ -2,149 +2,130 @@
+):
+    section_summary_context = SectionSummaryContext(
+        "en",
+        test_section_summary_schema,
+        answer_store,
+        list_store,
+        progress_store,
+        {},
+        {},
+        current_location=Location(section_id="property-details-section"),
+        routing_path=mocker.MagicMock(),
+    )
+
+    single_section_context = section_summary_context()
+
+    assert_summary_context(single_section_context)
+
+
+def test_build_view_context_for_section_summary(
+    test_section_summary_schema, answer_store, list_store, progress_store, mocker
+):
+    summary_context = SectionSummaryContext(
+        "en",
+        test_section_summary_schema,
+        answer_store,
+        list_store,
+        progress_store,
+        {},
+        {},
+        current_location=Location(
+            section_id="property-details-section",
+            block_id="property-details-summary",
+        ),
+        routing_path=mocker.MagicMock(),
+    )
+    context = summary_context()
+
+    assert "summary" in context
+    assert_summary_context(context)
+    assert len(context["summary"]) == 6
+    assert "title" in context["summary"]
+
+
+@pytest.mark.parametrize(
+    "answers, location, dict_path, expected_title",
+    (
+        (
+            [Answer(answer_id="house-type-answer", value="Semi-detached")],
+            Location(section_id="house-details-section"),
+            ("summary", "title"),
+            "Household Summary - Semi-detached",
+        ),
+        (
+            [],
+            Location(section_id="property-details-section"),
+            ("summary", "page_title"),
+            "Custom section summary title",
+        ),
+        (
+            [Answer(answer_id="house-type-answer", value="Semi-detached")],
+            Location(section_id="house-details-section"),
+            ("summary", "page_title"),
+            "Household Summary - …",
+        ),
+        (
+            [Answer(answer_id="number-of-people-answer", value=3)],
+            Location(section_id="household-count-section"),
+            ("summary", "page_title"),
+            "… people live here",
+        ),
+        (
+            [],
+            Location(section_id="property-details-section"),
+            ("summary", "title"),
+            "Property Details Section",
+        ),
+    ),
+)
+def test_custom_section(
+    answers,
+    location,
+    dict_path,
+    expected_title,
+    test_section_summary_schema,
+    answer_store,
+    list_store,
+    progress_store,
+    mocker,
+):
+    for answer in answers:
+        answer_store.add_or_update(answer)
+
+    summary_context = SectionSummaryContext(
+        "en",
+        test_section_summary_schema,
+        answer_store,
+        list_store,
+        progress_store,
+        {},
+        {},
+        current_location=location,
+        routing_path=mocker.MagicMock(),
+    )
+    context = summary_context()
+    assert get_dict_address(context, dict_path) == expected_title
```

Comment:
I think `get_dict_address` should be removed. It is just adding unnecessary ambiguity.
You are always asserting the same title key for the summary content, so this could be simplified to: `assert context['summary'][title_key] == expected_title`, where `title_key` or `key_to_assert` is the value such as `"page_title"` etc.